mne_connectivity/spectral/epochs.py

```diff
@@ -189,6 +189,46 @@ def _prepare_connectivity(epoch_block, times_in, tmin, tmax,
             n_signals, indices_use, warn_times)
 
 
+def _check_rank_input(rank, data, indices):
+    """Check the rank argument is appropriate and compute rank if missing."""
+    sv_tol = 1e-10  # tolerance for non-zero singular val (rel to largest)
+    if rank is None:
+        rank = np.zeros((2, len(indices[0])), dtype=int)
+
+        if isinstance(data, BaseEpochs):
+            data_arr = data.get_data()
+        else:
+            data_arr = data
+
+        for group_i in range(2):  # seeds and targets
+            for con_i, con_idcs in enumerate(indices[group_i]):
+                con_idcs = con_idcs[con_idcs != -1]  # -1 is padded value
+                s = np.linalg.svd(data_arr[:, con_idcs], compute_uv=False)
+                rank[group_i][con_i] = np.min(
+                    [np.count_nonzero(epoch >= epoch[0] * sv_tol)
+                     for epoch in s])
```

Comment:
Why do you need to check for -1 padded value again in here? Should it be taken care of before it reaches this function?

Reply:
I see. Okay, so it's because we need to still pass in the full-padded version to this function. Can you just add a note here? I guess it's non-ideal since upstream we already did this check. 

Maybe candidate for future refactoring to simplify code.